File: lexer-java/src/main/java/basicplus/frontend/SemanticDiagnostic.java
// ============================================================
// SemanticDiagnostic.java
// Mensaje (error o aviso) emitido por el analizador semántico.
// ============================================================
package basicplus.frontend;

public final class SemanticDiagnostic {

    public enum Kind { ERROR, WARNING }

    public final Kind kind;
    public final String message;
    public final int line;
    public final int column;

    public SemanticDiagnostic(Kind kind, String message, int line, int column) {
        this.kind = kind;
        this.message = message;
        this.line = line;
        this.column = column;
    }

    @Override
    public String toString() {
        String label = (kind == Kind.ERROR) ? "error semántico" : "aviso semántico";
        return "[" + line + ":" + column + "] " + label + ": " + message;
    }
}

File: lexer/SemanticDiagnostic.cs
// ============================================================
// SemanticDiagnostic.cs
// Mensaje (error o aviso) emitido por el analizador semántico.
// ============================================================

namespace BasicPlus.Lexer;

public enum DiagnosticKind { Error, Warning }

public sealed class SemanticDiagnostic
{
    public DiagnosticKind Kind { get; }
    public string Message { get; }
    public int Line { get; }
    public int Column { get; }

    public SemanticDiagnostic(DiagnosticKind kind, string message, int line, int column)
    {
        Kind = kind; Message = message; Line = line; Column = column;
    }

    public override string ToString()
    {
        var label = Kind == DiagnosticKind.Error ? "error semántico" : "aviso semántico";
        return $"[{Line}:{Column}] {label}: {Message}";
    }
}
